tests/test_outcomes_api.py

```diff
@@ -294,3 +294,125 @@ def test_category_quantity_sum_querying(
             },
         ]
         assert data["data"] == correct_data
+
+
+class TestOutcomesAPI(BaseApiTest):
```

Comment:
I don't think we should test this change with a full integration test but with a specific unit test that stress validation.
- Failures on integration tests are generally tricky to troubleshoot. The test does not point you to the specific issue but it just tells you that something in the whole query processing + execution is wrong.
- You do not need to bear the full performance impact of running the query end to end just to test validation.
- You are not testing the invalid query case.